catroid/src/androidTest/java/org/catrobat/catroid/catrobattestrunner/CatrobatTestRunner.kt

```diff
@@ -0,0 +1,189 @@
+    var assetPath: String = ""
+
+    @JvmField
+    @Parameterized.Parameter(1)
+    var assetName: String = ""
+
+    companion object {
+        private const val TEST_ASSETS_ROOT = "catrobatTests"
+        private const val TIMEOUT = 10_000
+        private const val RETRIES = 5
+
+        @JvmStatic
+        @Parameterized.Parameters(name = "{0} - {1}")
+        @Throws(IOException::class)
+        fun data(): Iterable<Array<Any>> = getCatrobatAssetsFromPath(TEST_ASSETS_ROOT)
+
+        @Throws(IOException::class)
+        private fun getCatrobatAssetsFromPath(path: String): List<Array<Any>> {
+            val parameters: MutableList<Array<Any>> = ArrayList()
+            val assets = InstrumentationRegistry.getInstrumentation().context.assets.list(path)
+            if (null == assets) {
+                Assert.fail("Could not load assets")
+                return parameters
+            }
+            for (asset in assets) {
+                if (asset.endsWith(Constants.CATROBAT_EXTENSION)) {
+                    parameters.add(arrayOf(path, asset))
+                } else {
+                    parameters.addAll(getCatrobatAssetsFromPath("$path/$asset"))
+                }
+            }
+            return parameters
+        }
+    }
+
+    @Before
+    @Throws(Exception::class)
+    fun setUp() {
+        val projectName = assetName.replace(Constants.CATROBAT_EXTENSION, "")
+        TestUtils.deleteProjects(projectName)
+        FlavoredConstants.DEFAULT_ROOT_DIRECTORY.mkdir()
+        Constants.CACHE_DIR.mkdir()
+        val inputStream = InstrumentationRegistry.getInstrumentation().context.assets
+            .open("$assetPath/$assetName")
+        val projectArchive = StorageOperations
+            .copyStreamToDir(inputStream, Constants.CACHE_DIR, assetName)
+        Assert.assertTrue(
+            ProjectUnzipAndImportTask
+                .task(projectArchive)
+        )
+        val projectDir = File(FlavoredConstants.DEFAULT_ROOT_DIRECTORY, projectName)
+        Assert.assertTrue(
+            ProjectLoadTask
+                .task(projectDir, ApplicationProvider.getApplicationContext())
+        )
+    }
+
+    @After
+    @Throws(IOException::class)
+    fun tearDown() {
+        StorageOperations.deleteDir(Constants.CACHE_DIR)
+        TestUtils.deleteProjects(assetName.replace(Constants.CATROBAT_EXTENSION, ""))
+    }
+
+    @Test
+    @Throws(InterruptedException::class)
+    fun run() {
+        var result: ActivityResult? = null
+        val messages = StringBuilder()
+
+        run breaker@{
+            (1..repeat_test).forEach { runNr ->
+                baseActivityTestRule.launchActivity(null)
+                waitForReady()
+                result = baseActivityTestRule.activityResult
+                if (result?.resultCode == TestResult.STAGE_ACTIVITY_TEST_SUCCESS) {
+                    return@breaker
+                }
+                messages.append(String.format("Testrun Nr.: %d\n", runNr))
```

Comment:
Since this is an integer and doesn't rquire further formating you could use: 
```suggestion
                messages.append("Testrun Nr.: $runNr\n")) //or ${runNr + 1}
```